src/utils/streams.js

```diff
@@ -0,0 +1,304 @@
+      TRANSFORM, README, REVERSE])
+  .demandOption('action')
+  .alias('f', 'file')
+  .describe('f', 'Specify a file')
+  .alias('p', 'path')
+  .describe('p', 'Specify a path to css files')
+  .help('h')
+  .alias('h', 'help')
+  .version(false);
+
+const { argv } = yargs;
+
+const readmeObject = require('./readme');
+
+const _getStdinReadline = () => readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+  terminal: false,
+});
+
+const _reverseString = (str) => str.split('').reverse().join('');
+
+const _convertStringToUpperCase = (str) => str.toUpperCase();
+
+const _processStdin = (operation) => {
+  console.log('Please, type text for conversion:');
+  const readLine = _getStdinReadline();
+
+  readLine.on('line', (line) => {
+    console.log(operation(line));
+  });
+};
+
+const _validateStringParam = (param, errorMessage) => {
+  if (!param || typeof param !== 'string') {
+    console.log(errorMessage);
+    return false;
+  }
+
+  return true;
+};
+
+const _validateCsvFileExtension = (csvFileName, help) => {
+  if (path.extname(csvFileName) !== '.csv') {
+    console.log('ERROR. File should have ".csv" extension');
+    console.log(help);
+    return false;
+  }
+
+  return true;
+};
+
+const _outputFormattedMultilineJson = (jsonArray, writeStream) => {
+  writeStream.write('[\n  ');
+  writeStream.write(JSON.stringify(jsonArray[0]));
+  jsonArray.shift();
+  jsonArray.forEach((line) => {
+    writeStream.write(',\n  ');
+    writeStream.write(JSON.stringify(line));
+  });
+  writeStream.write('\n]\n');
+  writeStream.end();
+};
+
+const _convertFileFun = (fileName, writeStream, help) => {
+  csv()
+    .fromFile(fileName)
+    .then((jsonArray) => {
+      if (jsonArray.length === 0) {
+        writeStream.write('\n');
+        writeStream.end();
+      } else if (jsonArray.length === 1) {
+        writeStream.write(`${JSON.stringify(jsonArray[0])}\n`);
+        writeStream.end();
+      } else {
+        _outputFormattedMultilineJson(jsonArray, writeStream);
+      }
+    })
+    .catch((err) => {
+      console.log(err.message);
+      console.log(help);
+    });
+};
+
+const _getCssFilesForPath = (pathToCss) => {
+  try {
+    let files = fs.readdirSync(pathToCss);
+    files = files
+      .filter((file) => path.extname(file) === '.css')
+      .filter((file) => file !== 'bundle.css')
+      .map((file) => path.join(pathToCss, file));
+
+    if (files.length === 0) {
+      console.log('ERROR. There are no css files in the given directory.');
+      return [];
+    }
+
+    return files;
+  } catch (err) {
+    console.log(err.message);
+    return [];
+  }
+};
+
+const _removeFile = (fileName) => {
+  try {
+    fs.unlinkSync(fileName);
+  } catch (err) {
+    if (err.code === 'ENOENT') {
+      // Do nothing.
+    } else {
+      console.log(err.message);
+      return false;
+    }
+  }
+
+  return true;
+};
+
+const _writeAllFilesToBundleCss = (fileArray, bundleCssFile, help) => {
```

Comment:
As an alternative, you might use promise chains in order to achieve pretty much the same effect but without recursion:

```
const files = fs.readdirSync('./').filter(filename => filename.indexOf(".js") !== -1)
    .map(fileName => fs.createReadStream(path.join(__dirname, fileName)))
    .reduce((acc, stream, index, streams) => {
        return acc.then(() => new Promise(resolve => {
            const writeStream = fs.createWriteStream(path.join(__dirname, 'bundle.js'), { flags: 'a' });
            stream.pipe(writeStream);
            stream.on('end', resolve);
        }));
    }, Promise.resolve());
```

This approach seems to have smaller cognitive complexity (it may be my personal opinion).
Anyway, providing you this as an example for general awareness